mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java

```diff
@@ -94,34 +95,51 @@ public Attachment getAttachment(AttachmentId attachmentId) throws AttachmentNotF
             .block();
     }
 
+    @Override
+    public InputStream loadAttachmentContent(AttachmentId attachmentId) throws AttachmentNotFoundException, IOException {
+        return attachmentDAOV2.getAttachment(attachmentId)
+            .map(daoAttachment -> blobStore.read(blobStore.getDefaultBucketName(), daoAttachment.getBlobId()))
+            .switchIfEmpty(attachmentDAO.getAttachmentAsBytes(attachmentId).map(ByteArrayInputStream::new))
```

Comment:
why would we load it entirely in memory just to have an inputstream?

Reply:
Outdated code: this code had been migrated to V2...